### PR TITLE
[REVIEW] Batch Hessian - Part I

### PR DESCRIPTION
**Pull request by frapac**
This PR implements the full batch Hessian operation in `src/Polar`. 

*tl;dr* 
In the optimization algorithm, we need to evaluate the Hessian for the objective, and for the constraints.
- Before that PR, the Hessians for the objective and the constraints were evaluated separately, leading to a lot of duplicate operations. The result was quite inefficient.
- Now, this PR uses a single kernel to evaluate all the Hessians together, using a single GPU kernel.

*Details*
- add a new structure `HessianLagrangian` to evaluate the reduced Hessian in the reduced space
- add three new functions `network_operations`, `cost_penalty_ramping_constraints` and `bus_injection`, with associated kernels
- rename objective function in `src/Polar` (from `active_power_generation` to `cost_production`)
- update `ProxALEvaluators` to use directly `cost_penalty_ramping_constraints` (so now ProxAL's objective is fully computed with AutoDiff)

### CODE REVIEW

**Review by michel2323 — APPROVED** [review]

**Comment by frapac** [comment]
Closing this PR in favor of #179
